perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,838 @@
+
+    db_creation_status = _RDSParseDBInstanceStatus(response)
+    status_query_count = 1
+    while True:
+      if (db_creation_status == 'available'):
+        break
+
+      if (db_creation_status != 'creating' and
+          db_creation_status != 'modifying' and
+          db_creation_status != 'backing-up' and
+          db_creation_status != 'rebooting'):
+        raise DBStatusQueryError('Invalid status in DB creation response. '
+                                 ' stdout is\n%s, stderr is\n%s' % (
+                                     stdout, stderr))
+
+      status_query_count += 1
+      if status_query_count > DB_STATUS_QUERY_LIMIT:
+        raise DBStatusQueryError('DB creation timed-out, we have '
+                                 'waited at least %s * %s seconds.' % (
+                                     DB_STATUS_QUERY_INTERVAL,
+                                     DB_STATUS_QUERY_LIMIT))
+
+      logging.info('Querying db creation status, current state is %s, query '
+                   'count is %d', db_creation_status, status_query_count)
+      time.sleep(DB_STATUS_QUERY_INTERVAL)
+
+      stdout, stderr, _ = vm_util.IssueCommand(status_query_cmd)
+      response = json.loads(stdout)
+      db_creation_status = _RDSParseDBInstanceStatus(response)
+
+    # We are good now, db has been created. Now get the endpoint address.
+    # On RDS, you always connect with a DNS name, if you do that from a EC2 VM,
+    # that DNS name will be resolved to an internal IP address of the DB.
+    if 'DBInstance' in response:
+      vm.db_instance_address = response['DBInstance']['Endpoint']['Address']
+    else:
+      if 'DBInstances' in response:
+        vm.db_instance_address = \
+            response['DBInstances'][0]['Endpoint']['Address']
+
+    logging.info('Successfully created an RDS DB instance. Address is %s',
+                 vm.db_instance_address)
+    logging.info('Complete output is:\n %s', response)
+
+  def Cleanup(self, vm):
+    """Clean up RDS instances, cleanup the extra subnet created for the
+       creation of the RDS instance.
+
+    Args:
+      vm: The VM that was used as the test client, which also stores states
+          for clean-up.
+    """
+
+    # Now, we can delete the DB instance. vm.db_instance_id is the id to call.
+    # We need to keep querying the status of the deletion here before we let
+    # this go. RDS DB deletion takes some time to finish. And we have to
+    # wait until this DB is deleted before we proceed because this DB holds
+    # references to various other resources: subnet groups, subnets, vpc, etc.
+    delete_db_cmd = util.AWS_PREFIX + [
+        'rds',
+        'delete-db-instance',
+        '--db-instance-identifier', vm.db_instance_id,
+        '--skip-final-snapshot']
+
+    logging.info('Will call this command to delete db instance: \n %s',
+                 ' '.join(delete_db_cmd))
+
+    stdout, stderr, _ = vm_util.IssueCommand(delete_db_cmd)
```

Comment:
The stdout and stderr of the deletion command is being checked in the loop below. First thing in the loop is to do json.load(stdout), and if the status is 'deleting', we will then issue the query_db_status command, and keep waiting until the DB instance is gone. Otherwise, we break out of the loop and look for stderr.

The DB deletion command can legitimately fail with an stderr containing message "DB instance not found" (and the exact text of such message is documented, but the return code is not). So my code below will depend on such message from stderr for a definitive confirmation of DB deletion. If after a long period of wait the DB is still stuck in Deleting state, then I'd have to bail... but I will still try to delete as many other resources as possible in this function, and raise an error at the final line.

I've tweaked the code here a bit to improve error checking and throwing, also changed the logging.warn to logging.error()